### ISSUE TITLE
incorrect hit point intersection, when segment.center x is the same as polygon edge

### ISSUE THREAD

**Issue by zhelezkov** [legends2k/2d-fov#1]
Hi! I really liked your article and the source codes for that 2d-fov. I've found a bug in your code, I spend 2 days finding and fix, but I can't find a solution. Please help.
A test scene is:
```javascript     
scene = {
         polygons: [
             { coords: [ 96,72,120,72,120,96,96,96 ],
               colour: "128, 128, 128",
               stroke: 1,
               fill: true },

        ],
         observer: { loc: vec2.fromValues(50, 96),
                     dir: vec2.fromValues(-0.707106781186, 0.707106781186),
                     colour: observerColour },
         target: { loc: vec2.fromValues(500, 100),
                   dir: vec2.fromValues(-1, 0),
                   colour: targetColour }
     };
```
As you see, fov calculation is incorrect

**Comment by ckapop** [comment]
Is this the bug that produces the following? The arc on the circle goes away at certain rotation.

![image](https://cloud.githubusercontent.com/assets/5403009/21656307/9bfedd82-d28b-11e6-8866-e656b06c3109.png)

**Comment by zhelezkov** [comment]
@ckapop actually no, it produces this:
![image](https://cloud.githubusercontent.com/assets/5163918/21679433/18786ab2-d355-11e6-9950-a1f824b6c9af.png)

**Comment by legends2k** [comment]
@rsredsq Thanks for filing this subtle, hard-to-find bug, I had fun fixing it!

Fixed by bfed453.

**Comment by legends2k** [comment]
@ckapop Thanks for bringing this up!  The issue you are taking about happens due to marginal floating-point round off error :(  It would get fixed if you increase the global variable `epsilon`.  Despite trying to play around with it at length, I was unable to come up with a single value that would work on all machines and on all browsers.  Hmm.. need to improve my knowledge on floating-point numbers to handle this more scientifically.
